crawl-ref/source/cloud.cc

```diff
@@ -908,16 +908,6 @@ bool actor_cloud_immune(const actor &act, const cloud_struct &cloud)
     // Qazlalites get immunity to clouds.
     if (player && have_passive(passive_t::cloud_immunity))
```

Comment:
Since you're removing the last interesting branch here, I'd just write this as:

    return (player && ...);

But that's a style choice; feel free to disagree.